src/dcos_e2e/cluster.py

```diff
@@ -269,74 +301,95 @@ def wait_for_dcos_ee(
 
-        credentials = {
-            'uid': superuser_username,
-            'password': superuser_password,
-        }
+            # In order to fully replace this method one would need to have
+            # DC/OS checks for every HTTP endpoint exposed by Admin Router.
 
-        any_master = next(iter(self.masters))
-        config_result = any_master.run(
-            args=['cat', '/opt/mesosphere/etc/bootstrap-config.json'],
-        )
-        config = json.loads(config_result.stdout.decode())
-        ssl_enabled = config['ssl_enabled']
-
-        scheme = 'https://' if ssl_enabled else 'http://'
-        dcos_url = scheme + str(any_master.public_ip_address)
-        enterprise_session = EnterpriseApiSession(  # type: ignore
-            dcos_url=dcos_url,
-            masters=[str(n.public_ip_address) for n in self.masters],
-            slaves=[str(n.public_ip_address) for n in self.agents],
-            public_slaves=[
-                str(n.public_ip_address) for n in self.public_agents
-            ],
-            auth_user=DcosUser(credentials=credentials),
-        )
+            credentials = {
+                'uid': superuser_username,
+                'password': superuser_password,
+            }
 
-        if ssl_enabled:
-            response = enterprise_session.get(
-                # We wait for 10 minutes which is arbitrary but should
-                # be more than enough after all systemd units are healthy.
-                '/ca/dcos-ca.crt',
-                retry_timeout=60 * 10,
-                verify=False,
+            any_master = next(iter(self.masters))
+            config_result = any_master.run(
+                args=['cat', '/opt/mesosphere/etc/bootstrap-config.json'],
+            )
+            config = json.loads(config_result.stdout.decode())
+            ssl_enabled = config['ssl_enabled']
+
+            scheme = 'https://' if ssl_enabled else 'http://'
+            dcos_url = scheme + str(any_master.public_ip_address)
+            enterprise_session = EnterpriseApiSession(  # type: ignore
+                dcos_url=dcos_url,
+                masters=[str(n.public_ip_address) for n in self.masters],
+                slaves=[str(n.public_ip_address) for n in self.agents],
+                public_slaves=[
+                    str(n.public_ip_address) for n in self.public_agents
+                ],
+                auth_user=DcosUser(credentials=credentials),
             )
-            response.raise_for_status()
-            enterprise_session.set_ca_cert()
 
-        enterprise_session.wait_for_dcos()
+            if ssl_enabled:
+                try:
+                    response = enterprise_session.get(
+                        # We wait for 10 minutes which is arbitrary but should
+                        # be more than enough after all systemd units are
+                        # healthy.
+                        '/ca/dcos-ca.crt',
+                        retry_timeout=60 * 10,
```

Comment:
Can we just have this be an infinite timeout then?

Reply:
Then we don't need to catch the exception

Reply:
To be clear - as written, this means that our users can give `None` and still get a `DCOSTimeoutError`.